app/passport/views.py

```diff
@@ -0,0 +1,105 @@
+import json
+import uuid
+
+from django.conf import settings
+from django.http import Http404, JsonResponse
+from django.shortcuts import render
+
+import web3
+from dashboard.utils import get_web3
+from eth_account.messages import defunct_hash_message
+
+from .models import PassportRequest
+
+
+def index(request):
+    # setup
+    player = request.GET.get('coinbase')
+    network = request.GET.get('network')
+    if not request.user.is_authenticated:
```

Comment:
Just use the @login_required decorator instead ?

Reply:
I believe if is a json endpoint shouldn't use the decorator as the error will be not a json and unexpected for the endpoint called from js.